lib/Bio/EnsEMBL/DataCheck/Utils.pm

```diff
@@ -321,4 +331,56 @@ sub is_compara_ehive_db {
   return $helper->execute_single_result(-SQL =>$sql);
 }
 
+=item B<same_metavalue>
+
+same_metavalue($mca, $old_mca, $meta_key);
+
+Takes two MetaContainer adaptors (C<$mca> and C<$old_mca>) and compares the two
+respective meta_value(s) for the given C<$meta_key>. Returns 1 if the values are
+the same, 0 otherwise.
+
+=back
+
+=cut
+sub same_metavalue {
+  my ($mca, $old_mca, $meta_key) = @_;
+
+  my $cur_metavalue = $mca->single_value_by_key($meta_key);
+  my $old_metavalue = $old_mca->single_value_by_key($meta_key);
+
+  return ($cur_metavalue eq $old_metavalue); 
+}
+
+=item B<same_assembly>
+
+same_assembly($mca, $old_mca);
+
+Takes two MetaContainer adaptors (C<$mca> and C<$old_mca>) and returns 1 if the
+two databases have the same assembly, 0 otherwise.
+
+=back
+
+=cut
+sub same_assembly {
+  my ($mca, $old_mca) = @_;
+
+  return same_metavalue($mca, $old_mca, 'asembly.default');
```

Comment:
```suggestion
  return same_metavalue($mca, $old_mca, 'assembly.default');
```
This typo didn't generate any obvious errors, because if you ask for a meta_key that doesn't exist, the Perl API gives you an 'undef' value, rather than failing. I think the behaviour we want would be for same_metavalue to be false if either value is undefined, since in the absence of information, it seems safest to assume difference?

Reply:
Yep, thank you, well spotted!